x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportStartDataFrameTransformAction.java

```diff
@@ -133,7 +132,26 @@ protected void masterOperation(StartDataFrameTransformAction.Request request,
                             "Unable to start data frame transform [" + config.getId() +
                                 "] as it is in state [" + transformState.getTaskState()  + "]", RestStatus.CONFLICT));
                     } else {
-                        persistentTaskActionListener.onResponse(existingTask);
+                        // If the task already exists but is not assigned to a node, something is weird
+                        // return a failure that includes the current assignment explanation (if one exists)
+                        if (existingTask.isAssigned() == false) {
```

Comment:
This is a check we should have been making to begin with and not doing it causes a bug. If the task exists, is not started, and is NOT assigned, that is an issue. We should not have tried to wait for its allocation and then cancel it.

Instead, I am opting for trying to start it directly if it is allocated, if not, return an error saying that the allocation failed.